strawberry/resolvers.py

```diff
@@ -30,6 +31,49 @@ def convert_enums_to_values(field: FieldDefinition, result: Any) -> Any:
     return result
 
 
+def get_arguments(
+    field: FieldDefinition, kwargs: Dict[str, Any], source: Any, info: Any
+) -> Tuple[List[Any], Dict[str, Any]]:
+    actual_resolver = cast(Callable, field.base_resolver)
+
+    kwargs = convert_arguments(kwargs, field.arguments)
+
+    # the following code allows to omit info and root arguments
+    # by inspecting the original resolver arguments,
+    # if it asks for self, the source will be passed as first argument
+    # if it asks for root, the source it will be passed as kwarg
+    # if it asks for info, the info will be passed as kwarg
+
+    function_args = get_func_args(actual_resolver)
+
+    args = []
+
+    if "self" in function_args:
```

Comment:
Unlikely edge-case, but should this only apply if `self` is the first arg?

Reply:
fixed 😊